src/hist.jl

```diff
@@ -259,34 +272,42 @@ end
 append!{T,N}(h::AbstractHistogram{T,N}, vs::NTuple{N,AbstractVector}, wv::WeightVec) = append!(h, vs, values(wv))
 
 
+# Turn kwargs nbins into a type-stable tuple of integers:
+function _nbins_tuple{N}(vs::NTuple{N,AbstractVector}, nbins)
```

Comment:
Maybe it would be better to include the most natural dispatch-based code under a `if VERSION >= "0.6.0+XXXX"` block so that we can remove this hack in the future?